GlobalConstantsService/src/ParticleDataList.cc

```diff
@@ -0,0 +1,121 @@
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+#include "cetlib_except/exception.h"
+
+#include "Offline/GlobalConstantsService/inc/ParticleDataList.hh"
+#include "Offline/ConfigTools/inc/ConfigFileLookupPolicy.hh"
+#include "Offline/ConfigTools/inc/SimpleConfig.hh"
+#include "Offline/DataProducts/inc/PDGCode.hh"
+
+
+namespace mu2e {
+
+  // *************************************************************
+
+  ParticleDataList::ParticleDataList( SimpleConfig const& config ) {
+
+    ConfigFileLookupPolicy findConfig;
+    
+    std::string tableFilename = findConfig(
+                    config.getString("particleDataList.filename",
+                   "Offline/GlobalConstantsService/data/ParticleList.txt") );
+    if( tableFilename.empty() ) {
+      throw cet::exception("PARTICLELIST_NO_FILE") 
+        << "ParticleDataList: file name not found\n";
+    }
+
+    std::ifstream in(tableFilename.c_str());
+    if ( !in ) {
+      throw cet::exception("PARTICLELIST_OPEN_FAIL")
+          << "Unable to open particle data file: "
+          << tableFilename << "\n";
+    }
+
+    std::string line;
+    std::string str;
+    std::vector<std::string> words;
+    while ( std::getline(in,line) ) {
+      std::istringstream iss(line);
+      while (iss >> str) {
+        words.emplace_back(str);
+      }
+      if ( words.size() != 7 ) {
+        throw cet::exception("PARTICLELIST_BAD_LINE")
+          << "ParticleList: Not 7 words on line: "
+          << line << "\n";
+      }
+      int id = std::stoi(words[0]);
+      _list.try_emplace(id, id, words[1], words[2], 
+         std::stof(words[4]), std::stof(words[5]), std::stof(words[6]) );
+      _names.try_emplace(words[1],id);
+      if( words[2] != "none" ) _names.emplace(words[2],id);
+      if( words[3] != "none" ) _names.emplace(words[3],id);
+      words.clear();
```

Comment:
Do you also want to add the aliases here too?

Reply:
words[3] is the alias, so I'm not sure what you mean

Reply:
Duh.  So the one that's missing is lookup by displayname.  What's the argument for not supporting that?

Reply:
That's line 53, words[1]

Reply:
Thanks.  I was reading the shape of the code, not what it said.